### PR TITLE
Tutorial for Menu

### PR DESCRIPTION
**Pull request by Ibrahim2750mi**
Please Review
What to review:
- [x] Check any grammar or spell errors.
- [x] Check that each diff hunk in menu_0[0-9].py is mentioned in the respective step [0-9].
- [x] Check any blank lines at the end of the codeblocks
- [x] Missing punctuation marks and using 'I' instead of 'we'

~~Also currently waiting for @eruvanos to propose a fix for UIDropdown 🙏🏽 in `menu_05.py`.~~ ~~The code errors now when operation the UISlider, its because of a bug/wrong code style.~~

### CODE REVIEW

**Comment by Ibrahim2750mi** [comment]
The UISlider throwing an exception should be fixed now.

File: doc/tutorials/menu/index.rst [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+    :pyobject: MainView.on_hide_view
+
+We also need to draw the childrens of the menu in ``on_draw``.
+
+ .. literalinclude:: menu_02.py
+    :caption: Drawing Children's of the Manager
+    :pyobject: MainView.on_draw
+    :emphasize-lines: 6-7
+
+Now we have successfully setup the manager, only thing left it to add the button. We are using ``UIAnchorLayout`` to position the button. We also setup a function which is called when the button is clicked.
```
Comment (pvcraven) [inline doc/tutorials/menu/index.rst]: Could we keep lines to about 80 characters? I know this isn't standard everywhere but I really, really, like coding w/o word wrap and no line breaks makes this stuff hard to read.

File: doc/tutorials/menu/menu_01.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
+"""
+Menu.
+"""
```
Comment (pvcraven) [inline doc/tutorials/menu/menu_01.py:5]: Since this is included in text, I'd drop the header lines. Or if you want to keep them, expand this to be at least a little more descriptive.

File: doc/tutorials/menu/menu_01.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+    def on_draw(self):
+        """ Render the screen. """
+        # Clear the screen
+        self.clear()
+
+
+def main():
+    """ Main function """
+
```
Comment (pvcraven) [inline doc/tutorials/menu/menu_01.py]: For in-text examples, probably drop some of the extra comments like the ones above.

**Review by pvcraven — CHANGES_REQUESTED** [review]
Is this ready? You've got 1/4 tasks but the PR is not in a 'draft' state, so I'm a bit confused. If it is ready to be included, keep it in the current 'ready' state. Otherwise set to 'draft'.

I'm excited to get a tutorial for GUI in, and this looks good.

**Comment by Ibrahim2750mi** [comment]
> Is this ready? You've got 1/4 tasks but the PR is not in a 'draft' state, so I'm a bit confused. If it is ready to be included, keep it in the current 'ready' state. Otherwise set to 'draft'.

It is ready 100%, I have added those checkboxes for the people who choose to review just for double checking my stuff. I will tick those if you say so.

**Comment by Ibrahim2750mi** [comment]
@pvcraven  Also don't merge now @eruvanos said he will have a look sometime this week.

File: doc/tutorials/menu/index.rst (hunk 1 above)
Comment (Ibrahim2750mi) [reply doc/tutorials/menu/index.rst]: Updated in https://github.com/pythonarcade/arcade/pull/1608/commits/a93bb6e03fedd180accf50b4ab101e7caf9dac29

File: doc/tutorials/menu/menu_01.py (hunk 3 above)
Comment (Ibrahim2750mi) [reply doc/tutorials/menu/menu_01.py]: Updated in https://github.com/pythonarcade/arcade/pull/1608/commits/438501342233c8ff34f90907e395e29fc90d2d9e

File: doc/tutorials/menu/menu_01.py (hunk 2 above)
Comment (Ibrahim2750mi) [reply doc/tutorials/menu/menu_01.py:5]: Updated in https://github.com/pythonarcade/arcade/pull/1608/commits/438501342233c8ff34f90907e395e29fc90d2d9e

**Review by pvcraven — APPROVED** [review]

**Comment by pvcraven** [comment]
Looks good to me! I didn't merge because you said "Also don't merge now @eruvanos said he will have a look sometime this week."

**Comment by eruvanos** [comment]
I have basically the full day tomorrow, I plan to care about all PRs regarding GUI and finish my left over tasks.

**Comment by pushfoo** [comment]
**tl;dr I think this should be simplified and split, but we could do it in subsequent PRs**

There are multiple concerns mixed together in this PR which I think should be separate:
1. A tutorial on how to create a menu screen
2. Modal dialogs
3. Maximizing tutorial coverage of our UI widgets

I think we should consider something like the following steps within the framework:
1. Rethink our widget hierarchy a little, including making a `UINinePatchLayout` baseclass to encapsulate 9-patch and anchor layout behavior
2. Add a `class UIModal(UINinePatchLayout, UIMouseFilterMixin)`
3. Make `UIMessageBox` and this PR's `SubMenu` inherit from `UIModal`
4. Create a views index page to aggregate view & menu tutorials like the [shader tutorials index page](https://github.com/pythonarcade/arcade/blob/development/doc/tutorials/shader_tutorials.rst)
5. Split this PR's concerns into a Basic Menu & Advanced Menu tutorial

**Comment by eruvanos** [comment]
@pushfoo I get your concerns and agree, that we could split up the tutorial even more. 
I created a new issue for that: https://github.com/pythonarcade/arcade/issues/1667

Still I merge this and we can improve and split up things afterwards, if somebody can spend the time to do so.

@Ibrahim2750mi Thank you for your work!
